File: BuzzTalk.Data/Entities/Message.cs
using System;
using System.Collections.Generic;

namespace BuzzTalk.Data.Entities;

public partial class Message
{
    public int Id { get; set; }

    public int? FromId { get; set; }

    public int? ToId { get; set; }

    public string Content { get; set; } = null!;

    public DateTime? SentOn { get; set; }

    public bool IsRead { get; set; }

    public int? GroupId { get; set; }

    public virtual User? From { get; set; }

    public virtual Group? Group { get; set; }

    public virtual User? To { get; set; }
}

File: Chat/src/app/Models/Message.ts
export default interface Message {
  id?: number;
  content?: string;
  sentOn?: string; // DateTime in C# → string (ISO format) in TypeScript
  fromId?: number;
  isRead?: boolean;
  toId?: number|null;
  groupId?: number | null; // nullable int → optional or null
}
